FitTracker/app/src/main/java/com/pps/globant/fittracker/MainActivity.java

```diff
@@ -25,18 +25,8 @@ protected void onCreate(Bundle savedInstanceState) {
         ButterKnife.bind(this);
         callbackManager = CallbackManager.Factory.create();
         presenter = new LoginPresenter(new LoginModel(callbackManager, BusProvider.getInstance()), new LoginView(this, BusProvider.getInstance()));
-    }
-
-    @Override
-    protected void onResume() {
-        super.onResume();
-        BusProvider.register(presenter);
-    }
-
-    @Override
-    protected void onPause() {
-        super.onPause();
-        BusProvider.unregister(presenter);
+        BusProvider.register(presenter, presenter.getModel());
```

Comment:
You should move this into presenter, and call it since onResume() activity method